pkg/cloud/azureprovider.go

```diff
@@ -149,6 +151,64 @@ func getRegions(service string, subscriptionsClient subscriptions.Client, provid
 	}
 }
 
+func getRetailPrice(region string, skuName string, currencyCode string) (string, error) {
+	pricingURL := "https://prices.azure.com/api/retail/prices?$skip=0&currencyCode='" + currencyCode + "'"
+	filterBase := "&$filter="
+	var filterParams []string
+
+	if region != "" {
+		regionParam := "armRegionName eq '" + region + "'"
+		filterParams = append(filterParams, regionParam)
+	}
+
+	if skuName != "" {
+		skuNameParam := "armSkuName eq '" + skuName + "'"
+		filterParams = append(filterParams, skuNameParam)
+	}
+
+	if len(filterParams) > 0 {
+		filterParamsEscaped := url.QueryEscape(strings.Join(filterParams[:], " and "))
+		pricingURL += filterBase + filterParamsEscaped
+	}
+
+	klog.V(4).Infof("starting download retail price payload from \"%s\"", pricingURL)
+	resp, err := http.Get(pricingURL)
+
+	if err != nil {
+		klog.V(4).Infof("Bogus fetch of \"%s\": %v", pricingURL, err)
+		return "", err
+	}
+
+	pricingPayload := AzureRetailPricing{}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		klog.V(4).Infof("Error getting response: %v", err)
+	}
+
+	jsonErr := json.Unmarshal(body, &pricingPayload)
+	if jsonErr != nil {
+		klog.V(4).Infof("Error unmarshalling data: %v", jsonErr)
+	}
+
+	retailPrice := ""
+
+	for i := range pricingPayload.Items {
+		if !(pricingPayload.Items[i].Type != "Consumption" || strings.Contains(pricingPayload.Items[i].ProductName, "Windows")) {
+			retailPrice = fmt.Sprintf("%f", pricingPayload.Items[i].RetailPrice)
+		}
+	}
+
+	klog.V(4).Infof("done parsing retail price payload from \"%s\"\n", pricingURL)
```

Comment:
Propose we drop this log, or at least use the `pkg/log` deduped functions (e.g. `log.DedupedInfof(10, ...)`)